ecgtools/parsers/utilities.py

```diff
@@ -16,3 +23,53 @@ def extract_attr_with_regex(
         return match
     else:
         return None
+
+
+def reverse_filename_format(filename, templates):
+    """
+    Uses intake's ``reverse_format`` utility to reverse the string method format.
+    Given format_string and resolved_string, find arguments
+    that would give format_string.format(arguments) == resolved_string
+    """
+    x = {}
+
+    for template in templates:
+        try:
+            x = reverse_format(template, filename)
+            if x:
+                break
+        except ValueError:
+            continue
+    if not x:
+        print(f'Failed to parse file: {filename} using patterns: {templates}')
+    return x
+
+
+@lru_cache(maxsize=None)
+def get_asset_list(root_path, depth=0, extension='*.nc'):
+    from dask.diagnostics import ProgressBar
+
+    root = pathlib.Path(root_path)
+    pattern = '*/' * (depth + 1)
+
+    dirs = [x for x in root.glob(pattern) if x.is_dir()]
+
+    @dask.delayed
+    def _file_dir_files(directory):
+        try:
+            cmd = ['find', '-L', directory.as_posix(), '-name', extension]
+            proc = subprocess.Popen(cmd, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+            output = proc.stdout.read().decode('utf-8').split()
+        except Exception:
+            output = []
+        return output
+
+    print('Getting list of assets...\n')
+    filelist = [_file_dir_files(directory) for directory in dirs]
+    # watch progress
+    with ProgressBar():
+        filelist = dask.compute(*filelist)
+
+    filelist = list(itertools.chain(*filelist))
+    print('\nDone...\n')
+    return filelist
```

Comment:
Is this necessary? i thought this gets taken care of by the `Builder` class

Reply:
Good point - it is not... we do not use it in the actual parsing function. I can go ahead and remove. Thanks for the other suggestions too!